### PR TITLE
add case sensitive flag to StringMatch

### PR DESCRIPTION
**Pull request by ramaraochavali**
Adds case sensitive flag to `StringMatch`.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
Once this is merged, I will follow-up with implementation.

File: networking/v1alpha3/virtual_service.proto [hunk 1]
```diff
@@ -907,6 +907,10 @@ message StringMatch {
     // ECMAscript style regex-based match
     string regex = 3;
   }
+
+   // Indicates that string matching should be case insensitive. The default is true.
```
Comment (rshriram) [inline networking/v1alpha3/virtual_service.proto]: is that the default today?

Comment (ramaraochavali) [reply networking/v1alpha3/virtual_service.proto]: Yes. If we do not specify, Envoy defaults it to true https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto.html?highlight=case_sensitive

Comment (ramaraochavali) [reply networking/v1alpha3/virtual_service.proto]: I realized we have `ignore_uri_case ` flag already defined that allow us to override the behaviour of uris. We are good with that. I will close this PR.

**Pull request closed without merging**